inspire/modules/authors/testsuite/test_author_record.py

```diff
@@ -0,0 +1,222 @@
+            <datafield tag="035" ind1="" ind2="">
+            <subfield code="9">authorid</subfield>
+            <subfield code="a">0000-0002-0003-2345</subfield></datafield>
+            <datafield tag="900" ind1="" ind2="">
+            <subfield code="a">9</subfield></datafield>
+            <datafield tag="100" ind1="" ind2="">
+            <subfield code="a">van der Nvenio, I.</subfield>
+            <subfield code="d">I like trains</subfield></datafield>
+            </record></collection>'''
+
+        record = self.create_record_from_marc(xml_string)
+
+        self.assertEqual(record['name']['birth_year'], None)
+        self.assertEqual(record['name']['death_year'], None)
+
+        validation = record.validate()
+        self.assertEqual(validation, {})
+
+    def test_inspire_identificators_types(self):
+        """Allow for ids from ORCID."""
+        json = {'ids': [{'type': 'orcid', 'value': '15783'},
+                {'type': 'authorid', 'value': 'somename'}],
+                'name': {'first': 'I.', 'last': 'Nvenio'},
+                'publications_list': [9]}
+
+        record = self.create_record_from_json(json)
+
+        validation = record.validate()
+        self.assertEqual(validation, {})
+
+    def test_inspire_wrong_identification_type(self):
+        """Don't allow unknown id types."""
+        json = {'ids': [{'type': 'sern', 'value': '15783'},
+                {'type': 'authorid', 'value': 'somename'}],
+                'name': {'first': 'I.', 'last': 'Nvenio'},
+                'publications_list': [9]}
+
+        record = self.create_record_from_json(json)
+
+        validation = record.validate()
+        self.assertTrue('type' in validation)
+
+
+class AuthorRecordOptionalFieldsTestCase(AuthorRecordTestCase):
+
+    """Tests for optional fields from author_base records."""
+
+    def test_optionality_of_fields(self):
+        """Some fields should not appear in the record.
+
+        It should happen when they were not provided in the input.
+        """
+        json = {'ids': [{'type': 'authorid', 'value': '15783'},
+                {'type': 'authorid', 'value': 'somename'}],
+                'name': {'first': 'I.', 'last': 'Nvenio'},
+                'publications_list': [9]}
+
+        record = self.create_record_from_json(json)
+
+        keys = record.keys()
+        self.assertFalse('emails' in keys)
+        self.assertFalse('other_names' in keys)
+        self.assertFalse('field_categories' in keys)
+        self.assertFalse('curators_note' in keys)
+        self.assertFalse('source' in keys)
+        self.assertFalse('prizes' in keys)
+        self.assertFalse('experiments' in keys)
+        self.assertFalse('phd_advisors' in keys)
+        self.assertFalse('conferences' in keys)
+        self.assertFalse('urls' in keys)
+        self.assertFalse('native_name' in keys)
+        self.assertFalse('date_added' in keys)
+        self.assertFalse('date_updated' in keys)
```

Comment:
In the same way, shouldnt we also test also that all fields that are required / that we define are actually there in `keys`?